packages/babel-register/test/integration.js

```diff
@@ -0,0 +1,67 @@
+import { exec as execCb } from "child_process";
+
+// TODO(Babel 8): Use util.promisify(execCb)
+const exec = (...args) =>
+  new Promise((resolve, reject) => {
+    execCb(...args, (error, stdout, stderr) => {
+      if (error) reject(error);
+      else resolve({ stdout, stderr });
+    });
+  });
+
+function fixture(name, file = "index.js") {
+  const cwd = `${__dirname}/fixtures/integration/${name}`;
+  return exec(`node -r ${__dirname}/.. ${cwd}/${file}`, {
+    cwd,
+    env: { ...process.env, BABEL_DISABLE_CACHE: true },
+  });
+}
+
+describe("integration tests", function () {
+  it("can hook into extensions defined by the config", async () => {
+    const { stdout, stderr } = await fixture("load-ts");
+
+    expect(stdout).toMatchInlineSnapshot(`
+      "LOADED: \\"<ROOT>/load-ts/index.js\\"
+      LOADED: \\"<ROOT>/load-ts/foo.ts\\"
+      DONE: foo.ts
+      DONE: index.js
+      "
+    `);
+    expect(stderr).toMatchInlineSnapshot(`""`);
+  });
```

Comment:
This test shows well the new feature :tada:
`@babel/register` can now load `.ts` files (or in general, any extension) by specifying it [in `babel.config.json`](https://github.com/babel/babel/pull/12151/files#diff-421ef23ba683a857425d217f1ddf8d30R2) rather than in the programmatic options